### PR TITLE
Add mural.js for opening and closing mural modals

### PR DESCRIPTION
**Pull request by Tara-Tiong**
This module includes open and close functions for mural modals. 
- One more debatable thing to add before merging: the `@media query` for the mural modal should be positioned better, where the bottom of the modal should stick at the edge of the screen 
- I'm unsure how to make the whole background stiff when the modal pops up. (if we should do this at all)
- I also don't know how to use the "data-mural-name" stuff
Screen Shot 2020-11-10 at 9 05 15 PM

# Mobile

Screen Shot 2020-11-10 at 9 06 01 PM

# Desktop

Screen Shot 2020-11-10 at 9 06 38 PM

### CODE REVIEW

File: scripts/index.js [hunk 1]
```diff
@@ -1 +1,2 @@
-import './header.js'
+import './header.js';
```
Comment (johncip) [inline scripts/index.js]: @Tara-Tiong @Ernesto2004 @kingtrentiii let's continue not using semicolons for JS:

https://blog.izs.me/2010/12/an-open-letter-to-javascript-leaders-regarding
http://inimino.org/~inimino/blog/javascript_semicolons

TLDR -- Javascript does something funny called "automatic semicolon insertion," and it does it even if you put semicolons in, so the ones you add yourself are useless 99% of the time -- and potentially confusing because they don't prevent ASI from taking place.

Like most style things, it's a topic of debate, but nowadays my view is that it's best to just use them only where they'd actually change the meaning, so that the code does exactly what it says (once you take JS's ASI behavior into account).

File: storybook/index.html [hunk 2]
```diff
@@ -9,7 +9,7 @@
             integrity="sha256-4+XzXVhsDmqanXGHaHvgh1gMQKX40OUvDEBTu8JcmNs="
             crossorigin="anonymous"></script>
         <script src="https://cdnjs.cloudflare.com/ajax/libs/ramda/0.25.0/ramda.min.js"></script>
-        <script src="/scripts/index.js" type="module" defer></script>
+        <script src="../scripts/index.js" type="module" defer></script>
```
Comment (johncip) [inline storybook/index.html]: the original line should work, as long as you're using a web server (like "go live" in vscode), but let me know if that's not the case

File: storybook/index.html [hunk 3]
```diff
@@ -374,13 +374,35 @@ <h1 class="sectionHeading">BAMP</h1>
...
 
                 <section class="demo" data-module-name="muralThumbnail">
-                    <button>
-                        <div class="box box-red">
-                            <div class="muralThumbnail--img"><img src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcS73Kk1K63Laa0bWG-7nwb7gCl4hKxoxeqfRw&usqp=CAU"></div>
-                            <div class="muralThumbnail--location">Oakland Ca</div>
-                            <div class="muralThumbnail--name">I dont Know</div>
+                    <div class="gridBox gridBox-col3 gridBox-center">
+                        <div class="muralBox">
+                            <button id="openMuralModal">
```
Comment (johncip) [inline storybook/index.html]: ID's can only be used once, so this will have to be a class -- and maybe we can just reuse the "muralThumbnail" class for targeting.

I think the ideal HTML would look like this:

```html
<button class="muralThumbnail box box-red" data-mural-name="lodestar">...</button>
```

...and then you could target it with

```javascript
document.querySelectorAll('.muralThumbnail')
```

File: storybook/index.html [hunk 4]
```diff
@@ -446,23 +468,47 @@ <h1 class="sectionHeading">BAMP</h1>
...
-                                <span class="muralModal--bold">Lead Artist</span><span class="muralModal--text">: Rachel</span>
-                            </p>
-                            <p>
-                              <img class="muralModal--icon" src="../images/map-marker.svg">
-                              <span class="muralModal--bold">Location</span><span class="muralModal--text">: Lodestar Public School</span>
-                            </p>
-                            <div class="muralModal--end">
-                                <img class="muralModal--icon" src="../images/calendar.svg">
-                                <span class="muralModal--bold">Date</span><span class="muralModal--text">: April, 2020</span>
+                    <div id="muralModalContainer" class="muralModalContainer">
```
Comment (johncip) [inline storybook/index.html]: same as above, we can't repeat IDs, and since we know we have multiple modals, we should just not use IDs at all for this.

I would also put the `data-mural-name` on the outer container div, like so:

```HTML
<div class="muralModalContainer" data-mural-name="lodestar">
    <diiv class="muralModal">...</div>
</div>
```

...the reason is that the container is what we'll be hiding and showing, so it will simplify the JS later

File: style/modules/muralModal.css [hunk 5]
```diff
@@ -57,13 +81,25 @@
         font-size: 17px;
     }
 
+    .muralModalContainer .muralModal {
```
Comment (johncip) [inline style/modules/muralModal.css:84]: let's remember to take a look at this later

File: scripts/mural.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+function showMuralModalContainer() {
+    const muralModalContainer = document.getElementById('muralModalContainer')
```
Comment (johncip) [inline scripts/mural.js]: see comments below... these will have to be targeted by class (since there are multiple), and you'll probably want to use `document.querySelectorAll` to grab them.

File: storybook/index.html [hunk 7]
```diff
@@ -374,13 +374,42 @@ <h1 class="sectionHeading">BAMP</h1>
...
 
                 <section class="demo" data-module-name="muralThumbnail">
-                    <button>
-                        <div class="box box-red">
-                            <div class="muralThumbnail--img"><img src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcS73Kk1K63Laa0bWG-7nwb7gCl4hKxoxeqfRw&usqp=CAU"></div>
-                            <div class="muralThumbnail--location">Oakland Ca</div>
-                            <div class="muralThumbnail--name">I dont Know</div>
+                    <div class="gridBox gridBox-col3 gridBox-center">
+                        <div>
+                            <!-- <button class="openMuralModal"> -->
```
Comment (johncip) [inline storybook/index.html]: we don't need the commented code anymore

File: style/modules/muralThumbnail.css [hunk 8]
```diff
@@ -4,6 +4,10 @@
     transition: opacity 0.5s ease;
 }
 
+#openMuralModal + #openMuralModal {
```
Comment (johncip) [inline style/modules/muralThumbnail.css]: I think this is unused code now that we're not using the ID?

File: style/modules/muralModal.css (hunk 5 above)
Comment (johncip) [reply style/modules/muralModal.css:84]: https://app.asana.com/0/1184406596339075/1199247589552151

**Review by johncip — APPROVED** [review]
nice! :shipit:
